### PR TITLE
Pyroscope: add public catalog description

### PR DESCRIPTION
**Pull request by bryanhuhta**
**What is this feature?**

Adds a README to the Pyroscope data source.

**Why do we need this feature?**

When searching the public catalog, it's important to show the Pyroscope data source since it's a built-in data source. ([slack](https://raintank-corp.slack.com/archives/C01C4K8DETW/p1692201366642309))

### CODE REVIEW

File: public/app/plugins/datasource/grafana-pyroscope-datasource/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Grafana Pyroscope is an open source continuous profiling platform. It will help you:
+* Find performance issues and bottlenecks in your code
+* Use high-cardinality tags/labels to analyze your application
+* Resolve issues with high CPU utilization
+* Track down memory leaks
+* Understand the call tree of your application
+* Auto-instrument your code to link profiling data to traces
+
+## [Live Demo](https://demo.pyroscope.io/)
```
Comment (bryanhuhta) [inline public/app/plugins/datasource/grafana-pyroscope-datasource/README.md:15]: Do we want to link to the demo here?

Comment (petethepig) [reply public/app/plugins/datasource/grafana-pyroscope-datasource/README.md:15]: I wish we had a new demo, but we don't have that right now, I think I would remove it.

Comment (petethepig) [reply public/app/plugins/datasource/grafana-pyroscope-datasource/README.md:15]: I'd maybe add a link to here: https://grafana.com/docs/grafana/latest/datasources/grafana-pyroscope/

File: public/app/plugins/datasource/grafana-pyroscope-datasource/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+[![Pyroscope GIF Demo](https://user-images.githubusercontent.com/23323466/143324845-16ff72df-231e-412d-bd0a-38ef2e09cba8.gif)](https://demo.pyroscope.io/)
```
Comment (petethepig) [inline public/app/plugins/datasource/grafana-pyroscope-datasource/README.md:17]:
```suggestion
```

**Review by petethepig — APPROVED** [review]
Looks good, but I would remove demo GIF and link

**Review by aocenas — APPROVED** [review]

**Pull request merged**